### DISCUSSION TITLE
Move back and forth in moves history

### DISCUSSION THREAD

**Discussion by Bolling88** [alexobviously/bishop#47] [Q&A] [answered]
I was wondering if there is an easy way to move back and forth with made moves to view the board in different states after the game has ended?

**Comment by alexobviously** [comment, marked as answer]
So the basic way is to use `Game.history` which contains all the states of the game in order.

Or you can use [GameNavigator](https://pub.dev/documentation/bishop/latest/bishop/GameNavigator-class.html)

I actually only just added this in the latest version, so it's not super well documented yet and the API might change slightly in the near future, but it does work for what you want.

Just pass your finished `Game` to it in the constructor and it will build the navigator for you. You can also build from a PGN.

Note that the states are missing information about checks (in both cases), but I'm in the middle of reworking that at the moment so expect that change fairly soon.
